cpp/include/raft/spatial/knn/detail/ann_utils.cuh

```diff
@@ -305,7 +310,7 @@ void accumulate_into_selected(size_t n_rows,
     case pointer_residency::host_only: {
       stream.synchronize();
       for (size_t i = 0; i < n_rows; i++) {
-        uint32_t l = row_ids[i];
+        uint32_t l = (uint32_t)row_ids[i];
```

Comment:
Consider not using C-style conversion (clang-tidy in our config should have generated a warning for this).